### PR TITLE
release/public-v2: remove WW3, update modulefiles and cmake configs, update .gitmodules

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

This PR:
- removes WW3 from the repository (not needed for SRW App release)
- updates `.gitmodules` to point to release/public-v2 branches of submodules
- updates most modulefiles (all machines I have access to) and updates/adds cmake configs as required
- fixes a bug in GFDL_atmos_cubed_sphere that caused b4b differences when using different threads or changing the MPI layout in regional configurations (from @chan-hoo)
- replace GSL drag suite with CIRES UGWP/GFS GWD in `tests/tests/fv3_ccpp_rrfs_v1beta` and `tests/tests/fv3_ccpp_rrfs_v1beta_debug` (a decision made by the SRW App Release Team leads)
- update `rt.conf` to run regression tests relevant for SRW App release, run on the following platforms: cheyenne.intel, cheyenne.gnu, hera.intel, hera.gnu, jet.intel, wcoss_cray, and possibly wcoss_dell_p3
- update `cmake/FindESMF.cmake` and `cmake/FindNetCDF.cmake` from develop (requested by @kgerheiser)
## Testing

New baselines created, tests passed and regression test logs updated on the following machines:
- cheyenne.gnu
- cheyenne.intel
- hera.gnu
- hera.intel
- jet.intel
- gaea.intel

## Dependencies

Associated PRs:

https://github.com/NOAA-EMC/NEMS/pull/80
https://github.com/NCAR/ccpp-framework/pull/324
https://github.com/NCAR/ccpp-physics/pull/500
https://github.com/NCAR/ccpp-physics/pull/501
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/54
https://github.com/NOAA-EMC/fv3atm/pull/173
https://github.com/ufs-community/ufs-weather-model/pull/205

### CODE REVIEW

**Review by DusanJovic-NOAA — APPROVED** [review]

File: cmake/configure_orion.intel.cmake [hunk 1]
```diff
@@ -1,2 +1 @@
 set(INLINE_POST     ON  CACHE BOOL "Enable inline post" FORCE)
```
Comment (junwang-noaa) [inline cmake/configure_orion.intel.cmake]: inline post is not available for regional fv3, maybe you don't need it compiled in the model as well

Comment (climbfuji) [reply cmake/configure_orion.intel.cmake]: > inline post is not available for regional fv3, maybe you don't need it compiled in the model as well

Good point.

Comment (climbfuji) [reply cmake/configure_orion.intel.cmake]: Done, removed it for both gnumake and cmake on all machines.

**Comment by climbfuji** [comment]
@junwang-noaa @DusanJovic-NOAA this PR is finally ready. I am running the last set of regression tests on gaea, but we should finalize the code review (many small changes specific for the SRW App release) in my opinion.

We could consider deleting `conf/*` since the gnumake build is not used in release/public-v2 at all.

**Review by junwang-noaa — APPROVED** [review]

**Comment by climbfuji** [comment]
@junwang-noaa @DusanJovic-NOAA I uploaded the missing RT logs for gaea - ready to merge

**Comment by climbfuji** [comment]
Thank you, @junwang-noaa, for merging the PR!

**Pull request merged**